smart-contracts/test/Lock/partialWithdraw.js

```diff
@@ -0,0 +1,79 @@
+
+const Units = require('ethereumjs-units')
+
+const deployLocks = require('../helpers/deployLocks')
+const Unlock = artifacts.require('../Unlock.sol')
+let unlock, locks
+
+contract('Lock', (accounts) => {
+  before(async () => {
+    unlock = await Unlock.deployed()
+    locks = await deployLocks(unlock)
+  })
+
+  describe('partialWithdraw', () => {
+    let owner, price, initialLockBalance, withdrawalAmount
+    price = Units.convert('0.01', 'eth', 'wei')
+    withdrawalAmount = Units.convert('0.005', 'eth', 'wei')
+
+    before(() => {
+      const purchases = [accounts[1], accounts[2]].map((account) => {
+        return locks['OWNED'].purchaseFor(account, '', {
+          value: price,
+          from: account
+        })
+      })
+      return Promise.all(purchases)
+        .then(() => {
+          return locks['OWNED'].owner.call()
+        })
+        .then((_owner) => {
+          owner = _owner
+        })
+    })
+
+    it('should only allow the owner to withdraw', () => {
+      assert.notEqual(owner, accounts[1]) // Making sure
+      return locks['OWNED'].partialWithdraw(withdrawalAmount, {
+        from: accounts[1]
+      }).catch(error => {
+        assert.equal(error.message, 'VM Exception while processing transaction: revert')
+      })
+    })
+
+    it('should fail if too much is withdrawn', () => {
```

Comment:
For completeness, maybe add `should fail if requesting partial withdraw of 0`

Reply:
yep.

Reply:
done.